client/task_runner.go

```diff
@@ -997,6 +1019,16 @@ func (r *TaskRunner) run() {
 	}
 }
 
+// cleanup calls Driver.Cleanup when a task is stopping. Errors are logged.
+func (r *TaskRunner) cleanup() {
+	if drv, err := r.createDriver(); err != nil {
+		r.logger.Printf("[WARN] client: error creating driver to cleanup resources: %v", err)
+	} else {
+		ctx := driver.NewExecContext(r.taskDir, r.alloc.ID)
+		drv.Cleanup(ctx, r.createdResources)
```

Comment:
Do something like: https://github.com/hashicorp/nomad/blob/master/client/task_runner.go#L1266 but do it based on recoverable errors